<root main_tree_to_execute="BehaviorTree">
    <!-- ////////// -->
    <BehaviorTree ID="BehaviorTree">
        <ReactiveFallback>
            <Condition ID="IsTirettePresente"/>
            <SequenceStar>
                <Action ID="InitialiserRobot"/>
                <ReactiveSequence>
                    <Action ID="AfficherLeScore" score="0"/>
                    <ReactiveFallback>
                        <SubTree ID="campJaune"/>
                        <SubTree ID="campViolet"/>
                    </ReactiveFallback>
                </ReactiveSequence>
            </SequenceStar>
        </ReactiveFallback>
    </BehaviorTree>
    <!-- ////////// -->
    <BehaviorTree ID="campJaune">
        <ReactiveSequence>
            <Inverter>
                <Condition ID="IsCampViolet"/>
            </Inverter>
            <Condition ID="IsNotTimeOut"/>
            <Action ID="ApproachObject" location="1;2;2"/>
        </ReactiveSequence>
    </BehaviorTree>
    <!-- ////////// -->
    <BehaviorTree ID="campViolet">
        <ReactiveSequence>
            <Condition ID="IsCampViolet"/>
            <Condition ID="IsNotTimeOut"/>
            <Action ID="ApproachObject" location="2;2;1"/>
        </ReactiveSequence>
    </BehaviorTree>
    <!-- ////////// -->
    <TreeNodesModel>
        <Action ID="AfficherLeScore">
            <input_port default="0" name="score">Le Score du Robot</input_port>
        </Action>
        <Action ID="ApproachObject">
            <input_port default="1;2;2" name="location">direction o√π aller</input_port>
        </Action>
        <Action ID="InitialiserRobot"/>
        <Condition ID="IsBoutonArretUrgence"/>
        <Condition ID="IsCampViolet"/>
        <Condition ID="IsNotTimeOut"/>
        <Condition ID="IsTirettePresente"/>
        <Action ID="ResterSurPlace"/>
        <SubTree ID="campJaune"/>
        <SubTree ID="campViolet"/>
    </TreeNodesModel>
    <!-- ////////// -->
</root>

